runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -4875,6 +4926,26 @@ J9::Z::TreeEvaluator::multianewArrayEvaluator(TR::Node * node, TR::CodeGenerator
    generateS390LabelInstruction(cg, TR::InstOpCode::label, node, oolJumpLabel);
    generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BRC, node, oolFailLabel);
 
+#if defined(TR_TARGET_64BIT)
+   if (isIndexableDataAddrPresent)
+      {
+      /* Populate dataAddr slot of 1st dimension array. Arrays of non-zero size
+       * use contiguous header layout while zero size arrays use discontiguous header layout.
+       */
+      cursor = generateS390LabelInstruction(cg, TR::InstOpCode::label, node, populateFirstDimDataAddrSlot);
+      iComment("populateFirstDimDataAddrSlot.");
+
+      generateRXInstruction(cg, TR::InstOpCode::LAY
+         , node
+         , temp3Reg
+         , generateS390MemoryReference(targetReg, temp1Reg, TR::Compiler->om.contiguousArrayHeaderSizeInBytes(), cg));
```

Comment:
What are the typical limit for `contiguousArrayHeaderSizeInBytes` ? Can we use LA instead of LAY to save two bytes ?

Reply:
Yup, we can. At the moment, max limit for `contiguousArrayHeaderSizeInBytes` is 24. So we should be ok using LA.